xclim/cli.py

```diff
@@ -0,0 +1,239 @@
+    else:
+        ctx.obj["xr_kwargs"].pop("combine", None)
+        ds = xr.open_dataset(arg, **ctx.obj["xr_kwargs"])
+    ctx.obj["input"] = ds
+    return ds
+
+
+def _get_output(ctx):
+    if "ds_out" not in ctx.obj:
+        dsin = _get_input(ctx)
+        ctx.obj["ds_out"] = dsin.drop_vars(dsin.data_vars)
+    return ctx.obj["ds_out"]
+
+
+def _process_indicator(indicator, ctx, **params):
+    click.echo(
+        f"Processing : {indicator.format({'long_name': indicator.long_name}, params)['long_name']}"
+    )
+    dsin = _get_input(ctx)
+    dsout = _get_output(ctx)
+
+    for key, val in params.items():
+        if (
+            indicator._sig.parameters[key].default is inspect._empty
+        ):  # a Dataset is expected
+            if key == "tas" and val is None and key not in dsin.data_vars:
+                # Special case for tas.
+                params["tas"] = atmos.tg(dsin.tasmin, dsin.tasmax)
+            else:
+                params[key] = dsin[
+                    val or key
+                ]  # Either a variable name was given or the key is the name
+
+    var = indicator(**params)
+    dsout = dsout.assign({var.name: var})
+    ctx.obj["ds_out"] = dsout
+
+
+def _create_command(name):
+    indicator = _get_indicator(name)
+    params = []
+    for name, param in indicator._sig.parameters.items():
+        params.append(
+            click.Option(
+                param_decls=["--" + name],
+                default=None if param.default is inspect._empty else param.default,
+                show_default=True,
+                help=indicator._parameters_doc.get(name),
+            )
+        )
+
+    @click.pass_context
+    def _process(ctx, **kwargs):
+        return _process_indicator(indicator, ctx, **kwargs)
+
+    return click.Command(
+        name,
+        callback=_process,
+        params=params,
+        help=indicator.abstract,
+        short_help=indicator.long_name,
+    )
+
+
+@click.command(short_help="List indicators.")
+@click.argument("module", required=False, nargs=-1)
+@click.option(
+    "-i", "--info", is_flag=True, help="Prints more details for each indicator."
+)
+def list(module, info):
+    """List all indicators in MODULE
+
+    If MODULE is ommitted, lists everything.
+    """
+    if len(module) == 0:
+        module = "all"
+    formatter = click.HelpFormatter()
+    formatter.write_heading("Listing all available indicators for computation.")
+    for xcmod in [atmos, land, seaIce]:
+        modname = xcmod.__name__.split(".")[-1]
+        if module == "all" or modname in module:
+            with formatter.section(
+                click.style("Indicators in module ", fg="blue")
+                + click.style(f"{modname}", fg="yellow")
+            ):
+                formatter.write_dl(
+                    [
+                        (
+                            click.style(name, fg="yellow")
+                            + (
+                                (" (" + ind.var_name + ")")
```

Comment:
Use f-strings.

Reply:
List comprehensions are for one liners. Use a proper for loop here.

Reply:
```suggestion
                                (f" ({ind.var_name})")
```